src/ansible_navigator/image_manager/puller.py

```diff
@@ -141,24 +159,35 @@ def prologue_stdout(self):
         print("Updating the execution environment")
         print("\u002d" * width)
 
+    def _generate_pull_command(self) -> List[str]:
+        """Generate the pull command.
+
+        :returns: The list of command parts
+        """
+        command_line = [self._container_engine, "pull"]
+        if isinstance(self._arguments, list):
+            for argument in self._arguments:
+                command_line.extend(shlex.split(argument))
+        command_line.append(self._image)
+        return command_line
+
     def pull_stdout(self):
-        """pull the image, print to stdout
+        """Pull the image, print to stdout.
+
+        ``podman`` writes errors to stdout and ``docker`` writes to stderr.
 
-        podman writes to stderr
-        docker writes to stdout
+        This allows us to capture, log and color the error from ``docker`` but in the case
+        of ``podman`` the error is only printed on the screen.
+
+        In both cases, stdout is not captured so the user can see the progress on the screen
+        as the pull is happening.
         """
         try:
-            if self._container_engine == "podman":
-                subprocess.run([self._container_engine, "pull", self._image], check=True)
-            elif self._container_engine == "docker":
-                subprocess.run(
-                    [self._container_engine, "pull", self._image],
-                    check=True,
-                    stderr=subprocess.PIPE,
-                )
-            else:
-                raise ValueError("Unknown container engine")
-
+            command_line = self._generate_pull_command()
```

Comment:
In the post processor we check if either podman or docker is installed https://github.com/ansible/ansible-navigator/blob/main/src/ansible_navigator/configuration_subsystem/navigator_post_processor.py#L213

What if both are not installed, based on the logic at the above line looks like the value will be still set to `auto`?
If yes maybe we can handle the error at L160 here in post processor.

Reply:
160 is gone now, the only logic here now is 

```
            stderr_pipe = subprocess.PIPE if self._container_engine == "docker" else None
```
So we can capture it's stderr and log it.  If we introduce a 3rd container engine, we will treat it like we do podman.